extensions/google-java/src/main/kotlin/com/connectrpc/extensions/GoogleJavaProtoAdapter.kt

```diff
@@ -17,56 +17,49 @@ package com.connectrpc.extensions
 import com.connectrpc.CODEC_NAME_PROTO
 import com.connectrpc.Codec
 import com.google.protobuf.CodedOutputStream
-import com.google.protobuf.GeneratedMessageV3
+import com.google.protobuf.ExtensionRegistryLite
 import com.google.protobuf.Internal
+import com.google.protobuf.MessageLite
 import okio.Buffer
 import okio.BufferedSource
-import java.io.IOException
 import kotlin.reflect.KClass
+import kotlin.reflect.cast
 
 /**
  * Adapter to use Google's protobuf-java runtime for
  * deserializing and serializing data types.
  */
-internal class GoogleJavaProtoAdapter<E : GeneratedMessageV3>(
-    clazz: KClass<E>,
+internal class GoogleJavaProtoAdapter<E : MessageLite>(
```

Comment:
Even though this is the java runtime extension, not javalite, I left the signatures here as `MessageLite` and `ExtensionRegistryLite` to maximize the applicability. `Message` and `ExtensionRegistry` in the non-lite runtime _extend_ `MessageLite` and `ExtensionRegistryLite` respectively, so it's safe. And we don't actually need the types to provide any API beyond what `MessageLite` and `ExtensionRegistryLite` provide.